src/Build/Definition/Project.cs

```diff
@@ -2479,6 +2479,10 @@ private List<GlobResult> GetAllGlobs(List<ProjectItemElement> projectItemElement
 
                     if (!string.IsNullOrEmpty(itemElement.Include))
                     {
+                        if (NativeMethodsShared.IsWindows && FileUtilities.InvalidFileNameChars.Any(c => itemElement.Include.Contains(c)))
```

Comment:
I think it is a wrong location to scan the invalid path. It will end up to scan lots of itemSpec which are not globbing candidate at all. It should be moved into BuildGlobResultFrom???Item method, after it picks up glob fragments. For two reasons:
1, there are far less globbing items than none globbing items, it is much more efficient to filter out other items first, instead of filtering invalid path first (which yield far less positive hit.)
2, when one include contains multiple items, maybe one of them is invalid, but this would exclude all of them. It is way too aggressive to be correct.

Reply:
You mean about [here](https://github.com/microsoft/msbuild/blob/master/src/Build/Definition/Project.cs#L2509)?